Data/ByteString/Builder/RealFloat.hs

```diff
@@ -0,0 +1,227 @@
+-- |
+-- Module      : Data.ByteString.Builder.RealFloat
+-- Copyright   : (c) Lawrence Wu 2021
+-- License     : BSD-style
+-- Maintainer  : lawrencejwu@gmail.com
+--
+-- Floating point formatting for Bytestring.Builder
+--
+-- This module primarily exposes `floatDec` and `doubleDec` which do the
+-- equivalent of converting through `string7 . show`.
+--
+-- It also exposes `formatFloat` and `formatDouble` with a similar API as
+-- `GHC.Float.formatRealFloat`.
+--
+-- NB: this implementation matches `show`'s output (specifically with respect
+-- to default rounding and length). In particular, there are boundary cases
+-- where the closest and 'shortest' string representations are not used.
+-- Mentions of 'shortest' in the docs below are with this caveat.
+
+module Data.ByteString.Builder.RealFloat
+  ( FloatFormat
+  , floatDec
+  , doubleDec
+  , fixed
+  , fixedDefaultPrecision
+  , scientific
+  , generic
+  , formatFloat
+  , formatDouble
+  ) where
+
+import Data.ByteString.Builder.Internal (Builder)
+import qualified Data.ByteString.Builder.RealFloat.Internal as R
+import qualified Data.ByteString.Builder.RealFloat.F2S as RF
+import qualified Data.ByteString.Builder.RealFloat.D2S as RD
+import qualified Data.ByteString.Builder.Prim as BP
+import GHC.Float (roundTo)
+import GHC.Word (Word64)
+import GHC.Show (intToDigit)
+
+-- | Returns a rendered Float. Matches `show` in displaying in fixed or
+-- scientific notation
+{-# INLINABLE floatDec #-}
+floatDec :: Float -> Builder
+floatDec = formatFloat generic
+
+-- | Returns a rendered Double. Matches `show` in displaying in fixed or
+-- scientific notation
+{-# INLINABLE doubleDec #-}
+doubleDec :: Double -> Builder
+doubleDec = formatDouble generic
+
+-- | An opaque wrapper around `FloatFormat'` and an optional precision argument
+-- used by `formatFloat` and `formatDouble`.
+data FloatFormat = MkFloatFormat
+  { floatFormat' :: FloatFormat'
+  , precision :: Maybe Int
+  }
+
+-- | Standard notation with `n` decimal places
+fixed :: Int -> FloatFormat
+fixed n = MkFloatFormat FFixed (Just n)
+
+-- | Standard notation with the default precision (number of decimal places)
+fixedDefaultPrecision :: FloatFormat
+fixedDefaultPrecision = MkFloatFormat FFixed Nothing
```

Comment:
The number of the decimal places isn't truly "fixed", right? Maybe call it `decimal`?!

Reply:
`fixed` is from `FFFormat`. I renamed it all to `standard` notation but am open to alternatives if there's a more canonical name

Reply:
I was hoping to find a canonical name for this format but couldn't find one. `standard` seems good enough. If we find a better name later on, it would be easy to add it and to deprecate the old one.